### PR TITLE
feat: basic jwt support

### PR DESCRIPTION
**Pull request by tchardin**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,13 +20,15 @@ class Saturn {
...
       logURL: 'https://twb3qukm2i654i3tnvx36char40aymqq.lambda-url.us-west-2.on.aws/',
+      authURL: 'https://fz3dyeyxmebszwhuiky7vggmsu0rlkoy.lambda-url.us-west-2.on.aws/',
       connectTimeout: 5_000,
       downloadTimeout: 0
     }, opts)
 
     this.logs = []
     this.reportingLogs = process?.env?.NODE_ENV !== 'development'
     this.hasPerformanceAPI = typeof window !== 'undefined' && window?.performance
+    this.isBrowser = typeof window !== 'undefined'
```
[the remark this answers is not in the document]
Comment (DiegoRBaquero) [reply src/index.js:31]: This is good enough

**Review by DiegoRBaquero — APPROVED** [review]

File: src/index.js [hunk 2]
```diff
@@ -3,11 +3,13 @@ import { CID } from 'multiformats'
 import { extractVerifiedContent } from './utils/car.js'
 import { asAsyncIterable, asyncIteratorToBuffer } from './utils/itr.js'
 import { randomUUID } from './utils/uuid.js'
+import { getJWT } from './utils/jwt.js'
 
 class Saturn {
   /**
    *
    * @param {object} [opts={}]
+   * @param {string} [opts.clientKey]
```
[the remark this answers is not in the document]
Comment (guanzo) [reply src/index.js:12]: hmm, if its required then yeah.

File: src/index.js [hunk 3]
```diff
@@ -60,6 +69,12 @@ class Saturn {
       controller.abort()
     }, options.connectTimeout)
 
+    if (!this.isBrowser) {
+      options.headers = {
```
Comment (guanzo) [inline src/index.js:73]: Merge with existing `options.headers` if exists. I'm passing in headers from the arc widget.

**Review by guanzo — APPROVED** [review]

**Comment by guanzo** [comment]
tests failing due to new clientKey requirement.. just add random clientKey to our test clients?

**Comment by guanzo** [comment]
I just realized that test clients perform fetches, so the JWT fetching logic will be called as well.. is that ok?

**Pull request merged**